Framework/WorkflowAlgorithms/src/EQSANSMonitorTOF.cpp

```diff
@@ -52,22 +50,20 @@ void EQSANSMonitorTOF::exec() {
   // Get the monitor
   const std::vector<detid_t> monitor_list =
       inputWS->getInstrument()->getMonitors();
-  if (monitor_list.size() != 1) {
+  if (monitor_list.size() != 1)
     g_log.error() << "EQSANS workspace does not have exactly ones monitor! "
                      "This should not happen\n";
-  }
-  IDetector_const_sptr mon;
-  try {
-    mon = inputWS->getInstrument()->getDetector(monitor_list[0]);
```

Comment:
`monitor_list` contains detector IDs, `Instrument::getDetector` used, so you need `DetectorInfo` here, not `SpectrumInfo`. As above, use `indexOf` to convert ID to index.